### PR TITLE
Create profile 160060145

### PR DESCRIPTION
**Pull request by joaquinvillar**
History create profile 160060145 with specs controller model migrations and routes

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+class User < ApplicationRecord
+  validates :email, presence: true, uniqueness: true, format: \
+    { with: /(^[a-zA-Z0-9_.+\-]+@[a-zA-Z0-9\-]+\.[a-zA-Z0-9\-.]+$)/ }
+  validates :first_name, presence: true, length: { in: 6..60 }
```
Comment (acostami) [inline app/models/user.rb]: What about short names like Ana, Juan ? I would suggest just validating the presence.

File: app/models/user.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  validates :last_name, presence: true, length: { in: 6..60 }
```
Comment (acostami) [inline app/models/user.rb]: Idem above

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  validates :nickname, length: { in: 6..64 }, allow_nil: true, on: :create
```
Comment (acostami) [inline app/models/user.rb]: Why are we only validating on create? I think `on: :create` should be removed from all validations.

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+  validates :description, length: { maximum: 60 }, allow_nil: true, on: :create
```
Comment (acostami) [inline app/models/user.rb]: Why validating maximum?

File: app/models/user.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  validates :image, length: { maximum: 60 }, allow_nil: true, on: :create
```
Comment (acostami) [inline app/models/user.rb]: Why validating `maximum: 60`? I think it won't be enough if we are going to use s3 urls.

File: app/models/user.rb [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+  validates :address, length: { maximum: 60 }, allow_nil: true, on: :create
```
Comment (acostami) [inline app/models/user.rb]: Why validating maximum?

File: config/routes.rb [hunk 7]
```diff
@@ -1,4 +1,4 @@
 Rails.application.routes.draw do
-  # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
+  resources :users
```
Comment (acostami) [inline config/routes.rb]: This is adding all routes for users, and we only have one action on the controller. we can do something like:
```
resources :users, only: :create
```

File: app/models/user.rb (hunk 3 above)
Comment (joaquinvillar) [reply app/models/user.rb]: what i wanted to do is accept the field nil only on create. Following this history:
https://www.pivotaltracker.com/n/projects/2190808

File: app/models/user.rb (hunk 4 above)
Comment (manuel-saravia) [reply app/models/user.rb]: What about setting a maximum length of 250?

File: app/models/user.rb (hunk 6 above)
Comment (manuel-saravia) [reply app/models/user.rb]: Do you think it's unnecesary to validate maximum lengths? In this case we thought a max of 60 should be enough for an address

File: app/models/user.rb (hunk 3 above)
Comment (acostami) [reply app/models/user.rb]: I think this validation will only validate when creating a user but if I update it I can add a nickname of length 1 if I want. 
I think what you want to do is something like:
```
validates :nickname, length: { in: 6..64 }, on: :create
validates :nickname, length: { in: 6..64 }, presence: true, on: :update
```

**Pull request merged**